lib/daru/index.rb

```diff
@@ -84,6 +84,66 @@ def [](key, *rest)
       end
     end
 
+    # Returns true if all arguments are either a valid category or position
+    # @param [Array<object>] *indexes categories or positions
+    # @return [true, false]
+    # @example
+    #   idx.respond? :a, 2
+    #   # => true
+    #   idx.respond? 3
+    #   # => false
+    def respond? *indexes
+      indexes.all? { |i| to_a.include?(i) || (i.is_a?(Numeric) && i < size) }
+    end
+
+    # Returns positions given indexes or positions
+    # @note If the arugent is both a valid index and a valid position,
+    #   it will treated as valid index
+    # @param [Array<object>] *indexes indexes or positions
+    # @example
+    #   x = Daru::Index.new [:a, :b, :c]
+    #   x.pos :a, 1
+    #   # => [0, 1]
+    def pos *indexes
+      indexes = preprocess_range(indexes.first) if indexes.first.is_a? Range
+
+      if indexes.size == 1
+        self[indexes.first]
+      else
+        indexes.map { |index| by_single_key index }
+      end
+    end
+
+    def subset *indexes
```

Comment:
To be honest, I'm not sure about this method still.
In current codebase, it is used only 2 times, and both of them are wanting to use _strictly_ positions list (not a range, not indexes list), and if I'm not missing something, could be replaced with `index.at(*positions)`. 

I'm not a big fan of methods able to do several totally different things based on subtle arguments change (especially those things like when you have the same arguments, but different behavior):
```ruby
i1 = Daru::Index.new [1,2,4]
i2 = Daru::Index.new [1,2,3]
i3 = Daru::Index.new [1,2,3,5,7]
i1.subset(4) # => #<Daru::Index(1): {4}>
i2.subset(4) # => #<Daru::Index(1): {}>
i3.subset(4) # => #<Daru::Index(1): {7}>
```
No code relying of this (and similar) method could be error-prone enough.

Are we really, totally, faithfully need this?